fabtests/functional/rdm_stress.c

```diff
@@ -593,58 +592,280 @@ static int run_child(void)
 
 	for (i = 0; i < ctrl_cnt && !ret; i++) {
 		printf("(%d-%d) rpc op %s\n", myid, id_at_server,
-		       rpc_op_str(ctrl[i].op));
-		ret = ctrl_op[ctrl[i].op](&ctrl[i]);
+		       rpc_op_str(ctrls[i].op));
+		ret = ctrl_op[ctrls[i].op](&ctrls[i]);
 	}
 
 free:
 	ft_free_res();
 	return ret;
 }
 
+static bool get_uint64_val(const char *json_payload, jsmntok_t *token,
+			   uint64_t *val)
+{
+	if (token->type != JSMN_PRIMITIVE)
+		return false;
 
-static struct rpc_ctrl ctrl_array[] = {
-	{op_goodbye},
-	{op_hello},
-	{op_msg_req, 1000},
-	{op_msg_resp},
-	{op_tag_req, 2000},
-	{op_tag_resp},
-	{op_msg_req, 1000000},
-	{op_msg_resp},
-	{op_write_req, 5600000, {12000}},
-	{op_write_resp},
-	{op_read_req, 64000},
-	{op_read_resp},
-	{op_tag_req, 2000000},
-	{op_tag_resp},
-	{op_read_req, 32000},
-	{op_read_resp},
-	{op_write_req, 86000, {6000}},
-	{op_write_resp},
-	{op_read_req, 1000000},
-	{op_read_resp},
-	{op_write_req, 56000, {12000}},
-	{op_write_resp},
-	{op_sleep, 100},
-	{op_exit},
-};
+	return (sscanf(&json_payload[token->start], "%lu", val) == 1);
+}
 
-/* TODO: read and parse control file */
-static int init_rpc(void)
+static bool get_op_enum(const char *json_payload, jsmntok_t *token,
+			uint32_t *op)
 {
-	ctrl = ctrl_array;
-	ctrl_cnt = ARRAY_SIZE(ctrl_array);
+	const char *enum_val_str;
+	size_t enum_val_len;
```

Comment:
nit: I would just use str and len as variable names.  Fewer characters make the code easier to read and the shorter names don't impact understanding.  I'd also drop 'enum' from the function name.  How the op values are defined isn't important here.

Reply:
agreed